core/src/main/java/org/frankframework/pipes/IsUserInRolePipe.java

```diff
@@ -58,36 +61,49 @@ public void configure() throws ConfigurationException {
 		if (StringUtils.isNotEmpty(getNotInRoleForwardName())) {
 			notInRoleForward = findForward(getNotInRoleForwardName());
 			if (notInRoleForward==null) {
-				throw new ConfigurationException("notInRoleForwardName ["+getNotInRoleForwardName()+"] not found");
+				throw new ConfigurationException("notInRoleForwardName not found");
 			}
 		}
 	}
 
-	protected void assertUserIsInRole(PipeLineSession session, String role) throws SecurityException {
-		if (!session.isUserInRole(role)) {
-			throw new SecurityException("user is not in role ["+role+"]");
+	protected String inWhichRoleIsUser(PipeLineSession session, List<String> roles) {
+		ISecurityHandler securityHandler = session.getSecurityHandler();
+		return roles.stream()
+				.filter(securityHandler::isUserInRole)
+				.findFirst().orElse(null);
+	}
```

Comment:
The name of this method is not directly clear to me. It is not a check, it is also not returning a list. It returns the first matching role, right?
Perhaps something like: `getFirstMatchingUserRole` does better fit the meaning? Or perhaps you know something better.